gio/src/file.rs

```diff
@@ -1111,6 +1113,137 @@ pub trait FileExtManual: IsA<File> + Sized {
+            Self::Stringv(_) => FileAttributeType::Stringv,
+        }
+    }
+}
+
+impl<'a> ToGlibPtr<'a, glib::ffi::gpointer> for FileAttributeTypeValue<'a> {
+    type Storage = FileAttributeTypeValueStorage<'a>;
+
+    fn to_glib_none(&'a self) -> Stash<'a, glib::ffi::gpointer, Self> {
+        let s = match self {
+            Self::Pointer(_, pointer) => FileAttributeTypeValueStorage::Pointer(*pointer),
+            Self::String(value) => FileAttributeTypeValueStorage::String(
+                ToGlibPtr::<*mut libc::c_char>::to_glib_none(value).1,
+            ),
+            Self::ByteString(value) => FileAttributeTypeValueStorage::ByteString(
+                ToGlibPtr::<*mut libc::c_char>::to_glib_none(value).1,
+            ),
+            Self::Boolean(value) => FileAttributeTypeValueStorage::Boolean(value.into_glib()),
+            Self::Uint32(value) => FileAttributeTypeValueStorage::Uint32(value),
+            Self::Int32(value) => FileAttributeTypeValueStorage::Int32(value),
+            Self::Uint64(value) => FileAttributeTypeValueStorage::Uint64(value),
+            Self::Int64(value) => FileAttributeTypeValueStorage::Int64(value),
+            Self::Object(value) => {
+                FileAttributeTypeValueStorage::Object(ToGlibPtr::<
+                    <glib::Object as glib::translate::GlibPtrDefault>::GlibType,
+                >::to_glib_none(value))
+            }
+            Self::Stringv(value) => FileAttributeTypeValueStorage::Stringv(
+                ToGlibContainerFromSlice::<*mut *mut libc::c_char>::to_glib_none_from_slice(value)
+                    .1,
+            ),
+        };
+        Stash(s.as_ptr(), s)
+    }
+}
+
+impl FromGlib<(crate::ffi::GFileAttributeType, glib::ffi::gpointer)>
+    for FileAttributeTypeValue<'_>
+{
+    #[inline]
+    unsafe fn from_glib(value: (crate::ffi::GFileAttributeType, glib::ffi::gpointer)) -> Self {
+        FileAttributeTypeValue::Pointer(from_glib(value.0), value.1)
+    }
+}
+
+// checker-ignore-item
+pub enum FileAttributeTypeValueStorage<'a> {
+    Pointer(glib::ffi::gpointer),
+    String(<&'a str as ToGlibPtr<'a, *mut libc::c_char>>::Storage),
+    ByteString(<&'a str as ToGlibPtr<'a, *mut libc::c_char>>::Storage),
+    Boolean(glib::ffi::gboolean),
+    Uint32(&'a u32),
+    Int32(&'a i32),
+    Uint64(&'a u64),
+    Int64(&'a i64),
+    Object(Stash<'a, <glib::Object as glib::translate::GlibPtrDefault>::GlibType, glib::Object>),
+    Stringv(<&'a str as ToGlibContainerFromSlice<'a, *mut *mut libc::c_char>>::Storage),
+}
+
+impl FileAttributeTypeValueStorage<'_> {
+    pub fn as_ptr(&self) -> glib::ffi::gpointer {
+        match self {
+            Self::Pointer(s) => *s,
+            Self::String(s) => s.as_ptr() as _,
+            Self::ByteString(s) => s.as_ptr() as _,
+            Self::Boolean(s) => s as *const _ as _,
+            Self::Uint32(s) => *s as *const _ as _,
+            Self::Int32(s) => *s as *const _ as _,
```

Comment:
Some more types instead of `_` would make it clearer whether this is correct or not

```suggestion
            Self::Int32(s) => *s as *const i32 as _,
```

if that doesn't compile then the code is wrong